Wrapper/WrapperScreen.swift

```diff
@@ -0,0 +1,134 @@
+     Embeds a wrapped `Screen`'s backing view controller as a child view controller with a matching frame.
+
+     This `Screen` can be useful wen you'd like to add view controller functionality to an existing `Screen` when you
+     don't have access to the backing view controller. For example, you might use a `WrapperScreen` to present a share
+     sheet over an existing `Screen`.
+
+     `WrapperScreen` is intended for use with a `WrapperScreenViewController` subclass. `WrapperScreenViewController`
+     handles embedding the wrapped `Screen`, adding it's backing view controller as a child, and updating its frame to
+     match the the backing view, forwards child view controller methods, updates the wrapped screen when appropriate.
+
+     Provide a view controller description of your `WrapperScreenViewController` subclass in your conformance to
+     `WrapperScreen`:
+
+     ````
+     struct MyWrapper {
+         var wrapped: Content
+     }
+
+     extension ShareSheetScreen: WrapperScreen, Screen where Content: Screen {
+         public func viewControllerDescription(environment: ViewEnvironment) -> ViewControllerDescription {
+             MyWrapperScreenViewController<Content>.description(for: self, environment: environment)
+         }
+     }
+     ````
+
+     See `WrapperScreenViewController` for more information on how write your view controller subclass.
+     */
+    public protocol WrapperScreen: Screen {
+        /// The type of the `Screen` being wrapped.
+        associatedtype Wrapped: Screen
+
+        /// The `Screen` being wrapped.
+        var wrapped: Wrapped { get }
+    }
+
+    /**
+     The superclass for the backing view controller of a `WrapperScreen`.
+     */
+    open class WrapperScreenViewController<ScreenType: WrapperScreen>: ScreenViewController<ScreenType> {
+        let wrappedViewController: DescribedViewController
+
+        public required init(screen: ScreenType, environment: ViewEnvironment) {
+            self.wrappedViewController = DescribedViewController(screen: screen.wrapped, environment: environment)
+
+            super.init(screen: screen, environment: environment)
+
+            addChild(wrappedViewController)
+            wrappedViewController.didMove(toParent: self)
+        }
+
+        /// Subclasses should override this method in order to update any relevant UI bits when the screen model
+        /// changes.
+        ///
+        /// You must call `super.screenDidChange(from: previousScreen, previousEnvironment: previousEnvironment)` in
+        /// your implementation.
+        override open func screenDidChange(from previousScreen: ScreenType, previousEnvironment: ViewEnvironment) {
+            super.screenDidChange(from: previousScreen, previousEnvironment: previousEnvironment)
+
+            wrappedViewController.update(screen: screen.wrapped, environment: environment)
+        }
+
+        override open func loadView() {
+            view = UIView(frame: UIScreen.main.bounds)
```

Comment:
We should also apply this, as @kylebshr found:

```
        // Match the UIKit default for a view controllers view. Not doing so can cause unexpected layout behavior.
        view.autoresizingMask = [.flexibleWidth, .flexibleHeight]
```

Reply:
I do think this is worth doing, in most cases it doesn't affect Square code since we don't really _use_ autoresizing masks, but for an open source framework it seems good to do (and in case that ever changes at Square)

Reply:
- [ ] Set `autoresizingMask` on views